test/pkg/framework/kafka.go

```diff
@@ -72,3 +76,89 @@ func ScaleKafkaInstanceDown(namespace, kafkaInstanceName string) error {
 
 	return nil
 }
+
+// WaitForMessagesOnTopic waits for at least a certain number of messages are present on the given topic
+func WaitForMessagesOnTopic(namespace, kafkaInstanceName, topic string, numberOfMsg int, timeoutInMin int) error {
+	return WaitForOnOpenshift(namespace, fmt.Sprintf("%d message available on topic %s withing %d minutes", numberOfMsg, topic, timeoutInMin), timeoutInMin,
+		func() (bool, error) {
+			messages, err := GetMessagesOnTopic(namespace, kafkaInstanceName, topic)
+			if err != nil {
+				return false, err
+			}
+			GetLogger(namespace).Info(fmt.Sprintf("Got %d messages", len(messages)))
```

Comment:
Can you also print what messages were found? Can make a failure analysis easier.

Reply:
added